nativescript-angular/platform-providers.ts

```diff
@@ -66,7 +66,69 @@ export const defaultPageFactoryProvider = { provide: PAGE_FACTORY, useValue: def
 @Injectable()
 export class FrameService {
     // TODO: Add any methods that are needed to handle frame/page navigation
+    private frames: { frame: Frame, name: string, rootOutlet: string }[] = [];
+
+    // Return the topmost frame.
+    // TabView with frames scenario: topmost() will return the root TabViewItem frame,
+    // which could be the wrong topmost frame (modal with nested frame e.g.):
+    // TabViewItem -> Frame -> Modal -> Frame2 -> Frame2-Navigation
     getFrame(): Frame {
-        return topmost();
+        let topmostFrame = topmost();
+        const { cachedFrame, cachedFrameRootOutlet } = this.findFrame(topmostFrame);
+
+        if (cachedFrame && cachedFrameRootOutlet) {
+            const latestFrameByOutlet = this.getLatestFrameByOutlet(cachedFrameRootOutlet);
+
+            if (latestFrameByOutlet && latestFrameByOutlet !== cachedFrame) {
```

Comment:
What is the scenario where we do not have `latestFrameByOutlet` -- can we add it as a comment?

Reply:
Resolved